include/SFML/Graphics/Text.hpp

```diff
@@ -445,8 +541,10 @@ class SFML_GRAPHICS_API Text : public Drawable, public Transformable
     float                 m_letterSpacingFactor{1.f};                  //!< Spacing factor between letters
     float                 m_lineSpacingFactor{1.f};                    //!< Spacing factor between lines
     std::uint32_t         m_style{Regular};                            //!< Text style (see Style enum)
-    Color                 m_fillColor{Color::White};                   //!< Text fill color
-    Color                 m_outlineColor{Color::Black};                //!< Text outline color
+    std::vector<Color>    m_fillColor{Color::White};                   //!< Text fill color
+    Color                 m_underlineFillColor{Color::White};          //!< Text underline/strikethrough fill color
+    std::vector<Color>    m_outlineColor{Color::Black};                //!< Text outline color
+    Color                 m_underlineOutlineColor{Color::Black};       //!< Text underline/strikethrough outline color
```

Comment:
Do we expect `m_fillColor` and `m_outlineColor` to have the same size at all times? If so then perhaps we combine them into a single vector to better represent this invariant.